### DISCUSSION TITLE
In manualFilter I want to pass filterVariant in setColumnFilters

### DISCUSSION THREAD

**Discussion by masterbater** [KevinVandy/mantine-react-table#78] [Q&A] [answered]
```typescript
  {
        id: "client",
        accessorKey: "fullname",
        header: "Client",
        filterFn: "equals",
        filterVariant: "select",
        mantineFilterSelectProps: {
          data: clientList,
        },
      },
```
```typescript
 const [columnFilters, setColumnFilters] = useState<MRT_ColumnFiltersState>(
    []
  );
```
Hopefully it will return like this 
```typescript
[
    {
        "id": "client",
        "value": "5c35f40acf1e583f4a8a51d7"
        "filterVariant": "select", // Add this
    }
]
```
This is useful for detection the filterFn or filterVariant and make the backend process it base on it

**Comment by masterbater** [comment]
> ```ts
>   {
>         id: "client",
>         accessorKey: "fullname",
>         header: "Client",
>         filterFn: "equals",
>         filterVariant: "select",
>         mantineFilterSelectProps: {
>           data: clientList,
>         },
>       },
> ```
> 
> ```ts
>  const [columnFilters, setColumnFilters] = useState<MRT_ColumnFiltersState>(
>     []
>   );
> ```
> 
> Hopefully it will return like this
> 
> ```ts
> [
>     {
>         "id": "client",
>         "value": "5c35f40acf1e583f4a8a51d7"
>         "filterVariant": "select", // Add this
>     }
> ]
> ```
> 
> This is useful for detection the filterFn or filterVariant and make the backend process it base on it

Hopefully this will get supported but now I just make use of prisma dmmf

**Comment by KevinVandy** [comment, marked as answer]
All you have to do is read or manage the columnFilterFns state so this is already supported

**Reply by masterbater** [reply]
onColumnFiltersChange={setColumnFilters}
 I was using this to listen to changes in filter change. But this one 
 onColumnFilterFnsChange do exist. Sorry didnt catch this one thanks for taking your time answering me.
